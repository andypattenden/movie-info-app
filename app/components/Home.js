var React = require('react');
var Link = require('react-router-dom').Link;
var PropTypes = require('prop-types');
var api = require('../utils/api');

function MovieList(props) {
	return (
		<ul className='list-unstyled'>
		{props.movies.map(function(movie, index){
			return (
				<li key={movie.Title}>
					#{index + 1}
					<div className='row'>
						<div className='col-sm-1'>
							<img className='img-responsive' src={movie.Poster} />
						</div>
						<div className='col-sm-6'>
							{movie.Title} ({movie.Year})
						</div>
						<div className='col-sm-2'>
							Score: {movie.imdbRating}
						</div>
						<div className='col-sm-3'>
							<Link to='/details'>View Details</Link>
						</div>
					</div>
				</li>
			)
		})}
		</ul>
	)
}

MovieList.propTypes = {
	movies: PropTypes.array.isRequired
}

class Home extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			movies: null
		}
	}

	componentDidMount() {
		api
			.fetchTopMovies()
			.then((movies) => {
				this.setState(function(){
					return {
						movies: movies
					}
				})
			});
	}

	render() {
		return (
			<div>
				<h1 className='text-center'>Top 10 Rated Movies</h1>

				{
					!this.state.movies ?
						'Loading...'
					:
						<MovieList movies={this.state.movies} />
				}

			</div>
		)
	}
}

module.exports = Home;
