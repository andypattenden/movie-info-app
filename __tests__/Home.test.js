import React from 'react';
import Home from '../app/components/Home';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
	const tree = renderer.create(
		<Home />
	).toJSON();

	expect(tree).toMatchSnapshot();
});
