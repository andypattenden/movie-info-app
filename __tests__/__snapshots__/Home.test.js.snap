// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <h1
      className="text-center"
    >
      IMDB Top 10 Rated Movies
    </h1>
  </div>
  <p
    className="loading"
  >
    Loading top 10 list...
  </p>
</div>
`;
